src/coprocessor/codec/mysql/duration.rs

```diff
@@ -233,71 +334,80 @@ impl Duration {
     /// and 10:10:10.000000 round 0 -> 10:10:10
     pub fn round_frac(mut self, fsp: i8) -> Result<Self> {
         let fsp = check_fsp(fsp)?;
-        if fsp >= self.fsp {
-            self.fsp = fsp;
+        if fsp >= self.fsp() {
+            self.set_fsp(fsp as u8);
             return Ok(self);
         }
-        self.fsp = fsp;
-        let nanos = f64::from(self.dur.subsec_nanos())
-            / f64::from(10u32.pow(NANO_WIDTH - u32::from(self.fsp)));
-        let nanos = (nanos.round() as u32) * (10u32.pow(NANO_WIDTH - u32::from(self.fsp)));
-        self.dur = StdDuration::new(self.dur.as_secs(), nanos);
-        Ok(self)
+
+        let mut hour = self.hour();
+        let mut minute = self.minute();
+        let mut second = self.second();
+        let mut nano = self.nano();
+        let round = 10u64.pow(NANO_WIDTH - u32::from(fsp));
+        nano /= round / 10;
+        nano = if nano % 10 > 4 {
+            nano / 10 + 1
+        } else {
+            nano / 10
+        } * round;
+        dbg!(nano);
```

Comment:
Remove this.

Reply:
Sorry! I forgot to remove the debug macro...